### PR TITLE
Introduce release automation

### PR DESCRIPTION
**Pull request by tsusanka**
About half way through I have realized the autoupdater can make Github releases as well :facepalm:. However it felt a bit less flexible than [release-it](https://github.com/release-it/release-it) so I went through with it after all. I am 90% sure some JavaScript gurus ( @keraf :) ) could make it way better though so we can definitely iterate on this.

Anyway, this PR:

- Introduces [release-it](https://github.com/release-it/release-it).
- Adds a CI job that publishes signed binaries (from the codesign jobs) to staging.
- Creates GitHub release with the assets attached.
- Fetches changelog from a GitHub issue which is opened and labeled `Release Changelog`. This way @matejzak can write the changelog as he is used to and the changelog is appended automatically. I am fetching it stupidly without a token so it will get limited easily, on the other hand this job will run sporadically so it should be just fine.
- Adds semi-automatic way to bump versions throughout the packages (see docs/releases/versioning.md). Partially solves https://github.com/trezor/trezor-suite/issues/3335.

This might need some adjustments during the release process, but I have tested on my fork and it worked.

### CODE REVIEW

File: ci/release-it.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+github:
+  release: true
+  draft: true
+  preRelease: true
+  releaseName: "v${version}"
+  releaseNotes: node ./ci/scripts/fetch_changelog.js
+  assets:
+    - "*.AppImage"
+    - "*.dmg"
+    - "*.exe"
```
Comment (prusnak) [inline ci/release-it.yml]: Maybe we want to include the .asc files as well?

```suggestion
    - "*.AppImage"
    - "*.AppImage.asc"
    - "*.dmg"
    - "*.dmg.asc"
    - "*.exe"
    - "*.exe.asc"
```

File: ci/releases.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  only:
+    refs:
+      - codesign
+  when: manual
+  script:
+    - source ${STAGING_SUITE_DEPLOY_KEYFILE}
+    - mkdir -p packages/suite-web/build/static/desktop
+    - 'rsync --delete -va "${DESKTOP_APP_NAME}"-*.AppImage ./packages/suite-web/build/static/desktop || :'
+    - 'rsync --delete -va "${DESKTOP_APP_NAME}"-*.dmg ./packages/suite-web/build/static/desktop || :'
+    - 'rsync --delete -va "${DESKTOP_APP_NAME}"-*.exe ./packages/suite-web/build/static/desktop || :'
```
Comment (prusnak) [inline ci/releases.yml]: Let's upload .asc files as well:

```suggestion
    - 'rsync --delete -va "${DESKTOP_APP_NAME}"-*.AppImage ./packages/suite-web/build/static/desktop || :'
    - 'rsync --delete -va "${DESKTOP_APP_NAME}"-*.AppImage.asc ./packages/suite-web/build/static/desktop || :'
    - 'rsync --delete -va "${DESKTOP_APP_NAME}"-*.dmg ./packages/suite-web/build/static/desktop || :'
    - 'rsync --delete -va "${DESKTOP_APP_NAME}"-*.dmg.asc ./packages/suite-web/build/static/desktop || :'
    - 'rsync --delete -va "${DESKTOP_APP_NAME}"-*.exe ./packages/suite-web/build/static/desktop || :'
    - 'rsync --delete -va "${DESKTOP_APP_NAME}"-*.exe.asc ./packages/suite-web/build/static/desktop || :'
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: ci/releases.yml (hunk 2 above)
Comment (tsusanka) [reply ci/releases.yml]: Accepted, great stuff! I have just written "fixup!" manually in front of the commit which should work I guess.. 053c839

File: ci/release-it.yml (hunk 1 above)
Comment (tsusanka) [reply ci/release-it.yml]: Accepted, great stuff! I have just written "fixup!" manually in front of the commit which should work I guess.. dbb4fd9

**Comment by tsusanka** [comment]
Ping @keraf, I would have this merged before Freeze :innocent:.

File: ci/scripts/fetch_changelog.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+// Fetches content of a first opened GitHub issue labeled "Release Changelog"
+
+const fetch = require("node-fetch");
+
+fetch("https://api.github.com/repos/trezor/trezor-suite/issues?labels=Release%20Changelog")
+.then(response => response.json())
+.then(data => {
+  console.log(data[0]["body"])
+})
+.catch(error => console.log("Changelog issue not found"))
```
Comment (keraf) [inline ci/scripts/fetch_changelog.js]:
``` 
const fetch = require('node-fetch');

fetch('https://api.github.com/repos/trezor/trezor-suite/issues?labels=Release%20Changelog')
    .then(response => response.json())
    .then(data => {
        console.log(data[0].body);
    })
    .catch(() => console.log('Changelog issue not found'));
```
Code is fine, just updated it to our style.

**Review by keraf — APPROVED** [review]

File: ci/scripts/fetch_changelog.js (hunk 3 above)
Comment (tsusanka) [reply ci/scripts/fetch_changelog.js]: ecae178

**Review by keraf — APPROVED** [review]

**Pull request merged**